app/models/species_list.rb

```diff
@@ -109,6 +109,18 @@ def remove_obs_callback(_obs)
     SiteData.update_contribution(:del, :species_list_entries, user_id)
   end
 
+  def clear
+    num = SpeciesList.connection.select_value(%(
+      SELECT COUNT(observation_id) FROM observations_species_lists
+      WHERE species_list_id = #{id}
+    ))
+    SpeciesList.connection.delete(%(
+      DELETE FROM observations_species_lists
+      WHERE species_list_id = #{id}
+    ))
```

Comment:
Will this work:
```suggestion
(SpeciesList.find(id)).observations.delete_all
```
It generates sql like this:
```sql
DELETE FROM `observations_species_lists` 
WHERE `observations_species_lists`.`species_list_id` = 75375684 
AND `observations_species_lists`.`observation_id` IN (573429226, 1046914674)
```

Reply:
I'm looking into this.  I'm concerned that it needs to include the full list of observation_ids explicitly.  At very least this will involve an additional query to look up those observation_ids (even thought it is completely unnecessary).  But I'm also concerned that this will trigger multiple callbacks.  I don't feel like this action should result in any callbacks.

Reply:
Doesn't look like it calls any callbacks, which is good.  But I'm still concerned that it has to load all the observations -- and it loads the *entire* observations, not just the ids.  Some species lists are huge.  It would be easy, in fact, to create a list with every observation on MO in it!

It's admittedly less "Railsy" but the SQL is so simple -- and it's a lot clearer to figure out what it actually does!  I don't have to spin up my VM and carefully test it to prove whether it calls callbacks or makes unnecessary expensive queries -- that I personally see no compelling reason not to use it.  It's not like some upgrade to Rails will ever invalidate basic low-level SQL!

Reply:
The SQL is best. (I can't find an efficient way to do this with an ActiveRecord query.)

Reply:
Including the rails version in a comment together with summary of this discussion seems to get us the best of both worlds.